planner/core/hints.go

```diff
@@ -295,6 +295,26 @@ func genHintsFromPhysicalPlan(p PhysicalPlan, nodeType nodeType) (res []*ast.Tab
 			Tables:   []ast.HintTable{{TableName: getTableName(index.Table.Name, index.TableAsName)}},
 			Indexes:  []model.CIStr{index.Index.Name},
 		})
+	case *PhysicalIndexMergeReader:
+		Indexs := make([]model.CIStr, 0, 2)
+		var tableName model.CIStr
+		var tableAsName *model.CIStr
+		for _, partialPlan := range pp.PartialPlans {
+			if index, ok := partialPlan[0].(*PhysicalIndexScan); ok {
+				Indexs = append(Indexs, index.Index.Name)
+				tableName = index.Table.Name
+				tableAsName = index.TableAsName
+			} else {
+				indexName := model.NewCIStr("pk")
```

Comment:
To be consistent with `show index`, we should use `PRIMARY` instead of `pk` as the index name?

Reply:
There is a problem. For example, in `select /*+ use_index_merge(t,primary,idxa) */ * form t where pk < 1 or a < 2`, there will throw an error at `primary` because it is a key word.